File: go/cnpj.go
package main

// Dabra - Consultar CNPJ em Go (stdlib)
// Uso: go run cnpj.go 00.000.000/0001-91
// Docs: https://dabradata.com/docs/receita-federal/receita-federal-pj
//
// Chave: variavel de ambiente DABRA_API_KEY. Com a chave de teste (dabra_test_...)
// a resposta e o exemplo do endpoint, a custo zero, com o header X-Example: true.

import (
	"encoding/json"
	"fmt"
	"io"
	"net/http"
	"net/url"
	"os"
	"time"
)

const baseURL = "https://app.dabradata.com/api/v1/consulta"

func apiKey() string {
	if k := os.Getenv("DABRA_API_KEY"); k != "" {
		return k
	}
	return "dabra_test_SUA_CHAVE"
}

// Receita Federal PJ (R$ 0,43): cadastro, CNAE, endereco e QSA.
func consultarCNPJ(cnpj string) (map[string]interface{}, error) {
	endpoint := baseURL + "/receita-federal-pj?" + url.Values{"cnpj": {cnpj}}.Encode()

	client := &http.Client{Timeout: 60 * time.Second}
	req, err := http.NewRequest("GET", endpoint, nil)
	if err != nil {
		return nil, err
	}
	req.Header.Set("X-API-Key", apiKey())

	resp, err := client.Do(req)
	if err != nil {
		return nil, err
	}
	defer resp.Body.Close()

	body, _ := io.ReadAll(resp.Body)
	if resp.StatusCode != 200 {
		return nil, fmt.Errorf("HTTP %d: %s", resp.StatusCode, body)
	}

	if resp.Header.Get("X-Example") == "true" {
		fmt.Println("(chave de teste: resposta de exemplo, sem custo)")
	}
	fmt.Printf("Custo: R$ %s | Saldo: R$ %s\n",
		resp.Header.Get("X-Request-Cost"),
		resp.Header.Get("X-Balance-Remaining"))

	var result map[string]interface{}
	if err := json.Unmarshal(body, &result); err != nil {
		return nil, err
	}
	return result, nil
}

func main() {
	cnpj := "00000000000191"
	if len(os.Args) > 1 {
		cnpj = os.Args[1]
	}

	dados, err := consultarCNPJ(cnpj)
	if err != nil {
		fmt.Fprintln(os.Stderr, "Erro:", err)
		os.Exit(1)
	}

	fmt.Printf("Razao social: %v\n", dados["razao_social"])
	fmt.Printf("Situacao:     %v\n", dados["descricao_situacao_cadastral"])
	fmt.Printf("CNAE:         %v\n", dados["cnae_fiscal_descricao"])
}

File: php/cnpj.php
<?php
// Dabra - Consultar CNPJ em PHP (7.4+, ext-curl)
// Docs: https://dabradata.com/docs/receita-federal/receita-federal-pj
//
// Chave: variavel de ambiente DABRA_API_KEY. Com a chave de teste (dabra_test_...)
// a resposta e o exemplo do endpoint, a custo zero, com o header X-Example: true.

$API_KEY  = getenv('DABRA_API_KEY') ?: 'dabra_test_SUA_CHAVE';
$BASE_URL = 'https://app.dabradata.com/api/v1/consulta';

// Receita Federal PJ (R$ 0,43): cadastro, CNAE, endereco e QSA.
function consultarCNPJ(string $cnpj, string $apiKey, string $baseUrl): array {
    $url = $baseUrl . '/receita-federal-pj?' . http_build_query(['cnpj' => $cnpj]);

    $ch = curl_init($url);
    curl_setopt_array($ch, [
        CURLOPT_RETURNTRANSFER => true,
        CURLOPT_HTTPHEADER     => ["X-API-Key: {$apiKey}"],
        CURLOPT_TIMEOUT        => 60,
        CURLOPT_HEADER         => true,
    ]);

    $response   = curl_exec($ch);
    $httpCode   = curl_getinfo($ch, CURLINFO_HTTP_CODE);
    $headerSize = curl_getinfo($ch, CURLINFO_HEADER_SIZE);
    curl_close($ch);

    $headers = substr($response, 0, $headerSize);
    $body    = json_decode(substr($response, $headerSize), true);

    if ($httpCode !== 200) {
        $erro = $body['error'] ?? [];
        throw new RuntimeException("HTTP {$httpCode} " . ($erro['code'] ?? '') . ': ' . ($erro['message'] ?? ''));
    }

    if (preg_match('/^X-Example:\s*true/mi', $headers)) {
        echo "(chave de teste: resposta de exemplo, sem custo)\n";
    }
    preg_match('/^X-Request-Cost:\s*([\d.]+)/mi', $headers, $cost);
    preg_match('/^X-Balance-Remaining:\s*([\d.-]+)/mi', $headers, $saldo);
    echo "Custo: R$ " . ($cost[1] ?? 'N/A') . " | Saldo: R$ " . ($saldo[1] ?? 'N/A') . "\n";

    return $body;
}

$dados = consultarCNPJ($argv[1] ?? '00.000.000/0001-91', $API_KEY, $BASE_URL);
echo "Razao social: " . $dados['razao_social'] . "\n";
echo "Situacao:     " . $dados['descricao_situacao_cadastral'] . "\n";
echo "CNAE:         " . $dados['cnae_fiscal_descricao'] . "\n";
